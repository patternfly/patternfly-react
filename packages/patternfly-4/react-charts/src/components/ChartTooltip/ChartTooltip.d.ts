import * as victory from 'victory';

export interface ChartTooltipProps extends victory.VictoryTooltipProps {
  themeColor?: string;
  themeVariant?: string;
}

declare const ChartTooltip: React.ComponentClass<ChartTooltipProps>;

export default ChartTooltip;
