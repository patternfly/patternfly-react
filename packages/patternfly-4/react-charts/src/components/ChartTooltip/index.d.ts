export { default as ChartTooltip, ChartTooltipProps } from './ChartTooltip';
