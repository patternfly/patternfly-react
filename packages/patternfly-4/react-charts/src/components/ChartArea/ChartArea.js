/* eslint-disable react/require-default-props,react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import hoistNonReactStatics from 'hoist-non-react-statics';
import { VictoryArea } from 'victory';
import { getTheme } from '../ChartTheme/themes/theme-utils';

export const propTypes = {
  /**
   * See TypeScript API docs: https://formidable.com/open-source/victory/docs/victory-area/
   */
  '': PropTypes.any,
  /*
   * Specifies the theme color; blue (default), green, or multi-color. Overridden by the theme property.
   */
  themeColor: PropTypes.string,
  /*
   * Specifies the theme variant; 'dark' or 'light' (default). Overridden by the theme property.
   */
  themeVariant: PropTypes.string
};

const ChartArea = ({
  themeColor,
  themeVariant,
  theme = getTheme(themeColor, themeVariant), // destructure last
  ...rest
}) => <VictoryArea theme={theme} {...rest} />;

// Note: VictoryArea.role must be hoisted
hoistNonReactStatics(ChartArea, VictoryArea);
ChartArea.propTypes = propTypes;

export default ChartArea;
