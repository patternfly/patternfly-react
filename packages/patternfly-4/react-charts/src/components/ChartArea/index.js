export { default as ChartArea } from './ChartArea';
