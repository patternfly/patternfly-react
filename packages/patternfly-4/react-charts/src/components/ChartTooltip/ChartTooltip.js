/* eslint-disable react/require-default-props,react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import hoistNonReactStatics from 'hoist-non-react-statics';
import { VictoryTooltip } from 'victory';
import { getTheme } from '../ChartTheme/themes/theme-utils';

export const propTypes = {
  /**
   * See TypeScript API docs: https://formidable.com/open-source/victory/docs/victory-tooltip/
   */
  '': PropTypes.any,
  /*
   * Specifies the theme color; blue (default), green, or multi-color. Overridden by the theme property.
   */
  themeColor: PropTypes.string,
  /*
   * Specifies the theme variant; 'dark' or 'light' (default). Overridden by the theme property.
   */
  themeVariant: PropTypes.string
};

const ChartTooltip = ({
  themeColor,
  themeVariant,
  theme = getTheme(themeColor, themeVariant), // destructure last
  ...rest
}) => <VictoryTooltip theme={theme} {...rest} />;

// Note: VictoryTooltip.defaultEvents must be hoisted
hoistNonReactStatics(ChartTooltip, VictoryTooltip);
ChartTooltip.propTypes = propTypes;

export default ChartTooltip;
