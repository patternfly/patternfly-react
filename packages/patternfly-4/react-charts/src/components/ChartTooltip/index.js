export { default as ChartTooltip } from './ChartTooltip';
