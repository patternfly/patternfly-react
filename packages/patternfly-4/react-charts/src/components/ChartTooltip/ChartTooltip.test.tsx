import * as React from 'react';
import { shallow } from 'enzyme';
import { ChartArea } from '../ChartArea/ChartArea';
import { ChartGroup } from '../ChartGroup/ChartGroup';
import { ChartVoronoiContainer } from '../ChartVoronoiContainer/ChartVoronoiContainer';
import { ChartTooltip } from './ChartTooltip';

Object.values([true, false]).forEach(isRead => {
  test(`ChartTooltip`, () => {
    const view = shallow(<ChartTooltip text="This is a tooltip" />);
    expect(view).toMatchSnapshot();
  });
});

test('allows tooltip via container component', () => {
  const view = shallow(
    <ChartGroup
      containerComponent={<ChartVoronoiContainer labels={(point) => "y: " + point.y} />}
      height={200}
      width={200}
    >
      <ChartArea
        data={[
          { name: 'Cats', x: 1, y: 1 },
          { name: 'Cats', x: 2, y: 2 },
          { name: 'Cats', x: 3, y: 3.2 },
          { name: 'Cats', x: 4, y: 3.5 }
        ]}
      />
      <ChartArea
        data={[
          { name: 'Dogs', x: 1, y: 0.5 },
          { name: 'Dogs', x: 2, y: 1 },
          { name: 'Dogs', x: 3, y: 2 },
          { name: 'Dogs', x: 4, y: 2.5 },
          { name: 'Dogs', x: 5, y: 2.5 }
        ]}
      />
    </ChartGroup>
  );
  expect(view).toMatchSnapshot();
});
