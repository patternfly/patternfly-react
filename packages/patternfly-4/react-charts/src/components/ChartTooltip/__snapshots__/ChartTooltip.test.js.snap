// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartTooltip 1`] = `
<VictoryTooltip
  active={false}
  flyoutComponent={
    <Flyout
      pathComponent={<Path />}
    />
  }
  groupComponent={<g />}
  labelComponent={
    <VictoryLabel
      capHeight={0.71}
      direction="inherit"
      lineHeight={1}
      textComponent={<Text />}
      tspanComponent={<TSpan />}
    />
  }
  renderInPortal={true}
  text="This is a tooltip"
  theme={
    Object {
      "area": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#bee1f4",
            "fillOpacity": 0.4,
            "stroke": "#73bcf7",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "axis": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "axis": Object {
            "fill": "transparent",
            "stroke": "#D1D1D1",
            "strokeLinecap": "round",
            "strokeLinejoin": "round",
            "strokeWidth": 1,
          },
          "axisLabel": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 100,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
          "grid": Object {
            "fill": "none",
            "pointerEvents": "painted",
            "stroke": "none",
          },
          "tickLabels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "ticks": Object {
            "fill": "transparent",
            "size": 1,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "bar": Object {
        "barWidth": 10,
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "stroke": "none",
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "boxplot": Object {
        "boxWidth": 20,
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "max": Object {
            "padding": 8,
            "stroke": "#73bcf7",
            "strokeWidth": 1,
          },
          "maxLabels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "median": Object {
            "padding": 8,
            "stroke": "#73bcf7",
            "strokeWidth": 1,
          },
          "medianLabels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "min": Object {
            "padding": 8,
            "stroke": "#73bcf7",
            "strokeWidth": 1,
          },
          "minLabels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "q1": Object {
            "fill": "#bee1f4",
            "padding": 8,
          },
          "q1Labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "q3": Object {
            "fill": "#bee1f4",
            "padding": 8,
          },
          "q3Labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "candlestick": Object {
        "candleColors": Object {
          "negative": "#73bcf7",
          "positive": "#bee1f4",
        },
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "stroke": "#73bcf7",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "chart": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "parent": Object {
            "border": "1px solid #bee1f4",
          },
        },
        "width": 450,
      },
      "errorbar": Object {
        "borderWidth": 8,
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#73bcf7",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "group": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "legend": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "gutter": 20,
        "orientation": "horizontal",
        "style": Object {
          "data": Object {
            "type": "square",
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "title": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 2,
            "stroke": "transparent",
          },
        },
        "titleOrientation": "top",
      },
      "line": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#73bcf7",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "pie": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 230,
        "padding": 8,
        "style": Object {
          "data": Object {
            "padding": 8,
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 8,
            "stroke": "#EDEDED",
          },
        },
        "width": 230,
      },
      "scatter": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#bee1f4",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "stack": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "stroke": "#fff",
            "strokeWidth": 1,
          },
        },
        "width": 450,
      },
      "tooltip": Object {
        "cornerRadius": 0,
        "flyoutStyle": Object {
          "cornerRadius": 0,
          "fill": "#151515",
          "pointerEvents": "none",
          "stroke": "#151515",
          "strokeWidth": 0,
        },
        "pointerLength": 10,
        "pointerWidth": 20,
        "style": Object {
          "fill": "#EDEDED",
          "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
          "fontSize": 14,
          "letterSpacing": "normal",
          "padding": 16,
          "pointerEvents": "none",
          "stroke": "#EDEDED",
          "textAnchor": "middle",
        },
      },
      "voronoi": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "flyout": Object {
            "fill": "#151515",
            "pointerEvents": "none",
            "stroke": "#151515",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 8,
            "pointerEvents": "none",
            "stroke": "#EDEDED",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
    }
  }
/>
`;

exports[`ChartTooltip 2`] = `
<VictoryTooltip
  active={false}
  flyoutComponent={
    <Flyout
      pathComponent={<Path />}
    />
  }
  groupComponent={<g />}
  labelComponent={
    <VictoryLabel
      capHeight={0.71}
      direction="inherit"
      lineHeight={1}
      textComponent={<Text />}
      tspanComponent={<TSpan />}
    />
  }
  renderInPortal={true}
  text="This is a tooltip"
  theme={
    Object {
      "area": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#bee1f4",
            "fillOpacity": 0.4,
            "stroke": "#73bcf7",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "axis": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "axis": Object {
            "fill": "transparent",
            "stroke": "#D1D1D1",
            "strokeLinecap": "round",
            "strokeLinejoin": "round",
            "strokeWidth": 1,
          },
          "axisLabel": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 100,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
          "grid": Object {
            "fill": "none",
            "pointerEvents": "painted",
            "stroke": "none",
          },
          "tickLabels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "ticks": Object {
            "fill": "transparent",
            "size": 1,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "bar": Object {
        "barWidth": 10,
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "stroke": "none",
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "boxplot": Object {
        "boxWidth": 20,
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "max": Object {
            "padding": 8,
            "stroke": "#73bcf7",
            "strokeWidth": 1,
          },
          "maxLabels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "median": Object {
            "padding": 8,
            "stroke": "#73bcf7",
            "strokeWidth": 1,
          },
          "medianLabels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "min": Object {
            "padding": 8,
            "stroke": "#73bcf7",
            "strokeWidth": 1,
          },
          "minLabels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "q1": Object {
            "fill": "#bee1f4",
            "padding": 8,
          },
          "q1Labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "q3": Object {
            "fill": "#bee1f4",
            "padding": 8,
          },
          "q3Labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "candlestick": Object {
        "candleColors": Object {
          "negative": "#73bcf7",
          "positive": "#bee1f4",
        },
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "stroke": "#73bcf7",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "chart": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "parent": Object {
            "border": "1px solid #bee1f4",
          },
        },
        "width": 450,
      },
      "errorbar": Object {
        "borderWidth": 8,
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#73bcf7",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "group": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "legend": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "gutter": 20,
        "orientation": "horizontal",
        "style": Object {
          "data": Object {
            "type": "square",
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "title": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 2,
            "stroke": "transparent",
          },
        },
        "titleOrientation": "top",
      },
      "line": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#73bcf7",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "pie": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 230,
        "padding": 8,
        "style": Object {
          "data": Object {
            "padding": 8,
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 8,
            "stroke": "#EDEDED",
          },
        },
        "width": 230,
      },
      "scatter": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#bee1f4",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "stack": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "stroke": "#fff",
            "strokeWidth": 1,
          },
        },
        "width": 450,
      },
      "tooltip": Object {
        "cornerRadius": 0,
        "flyoutStyle": Object {
          "cornerRadius": 0,
          "fill": "#151515",
          "pointerEvents": "none",
          "stroke": "#151515",
          "strokeWidth": 0,
        },
        "pointerLength": 10,
        "pointerWidth": 20,
        "style": Object {
          "fill": "#EDEDED",
          "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
          "fontSize": 14,
          "letterSpacing": "normal",
          "padding": 16,
          "pointerEvents": "none",
          "stroke": "#EDEDED",
          "textAnchor": "middle",
        },
      },
      "voronoi": Object {
        "colorScale": Array [
          "#06c",
          "#bee1f4",
          "#73bcf7",
          "#0066ff",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "flyout": Object {
            "fill": "#151515",
            "pointerEvents": "none",
            "stroke": "#151515",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#151515",
            "fontFamily": "overpass,overpass,open sans,-apple-system,blinkmacsystemfont,Segoe UI,roboto,Helvetica Neue,arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 8,
            "pointerEvents": "none",
            "stroke": "#EDEDED",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
    }
  }
/>
`;
